File: server/spring-boot-ecommerce/src/main/java/com/ecommerce/springbootecommerce/domain/Country.java
package com.ecommerce.springbootecommerce.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;



@Getter
@Setter
@Entity
@Table(name = "country")
public class Country {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private int id;

    private String name;

    private String code;

}

File: client/ecommerce-app/src/app/models/country.ts
interface CountryArgs {
  id?: number;
  country?: string;
  code?: string;
}

export class Country {
  id?: number;
  country?: string;
  code?: string;
  
  constructor(args: CountryArgs){
    this.id = args.id;
    this.country = args.country;
    this.code = args.code;
  }
}